Add certificate policy of type any policy id

RonEld opened the pull request:
## Description
Add a function for getting the certificate policy. Currently only
"Any Policy" is supported.
Added initial tests for oid module

## Status
**READY**

## Requires Backporting
NO  

## Additional comments
Any additional information that could be of interest

## Todos
- [x] Tests
- [x] Documentation
- [x] Changelog updated
- [ ] Backported

Patater commented on tests/suites/test_suite_oid.function:
```diff
@@ -0,0 +1,38 @@
+void oid_get_certificate_policies( char * oid_hex, char * result_str )
+{
+    unsigned char oid[64];
+    mbedtls_asn1_buf asn1_buf= { 0, 0, NULL };
```
Style: Add space before `=`

Patater commented on tests/suites/test_suite_oid.function:
```diff
@@ -0,0 +1,38 @@
+    size_t oid_len;
+    const char *desc;
+
+    oid_len = unhexify( oid, oid_hex );
```
Could you use a `data_t` instead of `char *` for the oid parameter? Then you don't have to unhexify or stack allocate 64 bytes, etc.

RonEld commented:
@Patater Thank you for your comments!

I have addressed your comments and added another test data, using a valid oid, but not for this type.

I have squashed the modifications to first commit, for quicker process, as these are minimal changes

Patater commented on tests/suites/test_suite_oid.function:
```diff
@@ -0,0 +1,35 @@
+    mbedtls_asn1_buf asn1_buf = { 0, 0, NULL };
+    int ret;
+    const char *desc;
+
```
Style: consider removing the extra newline here.

RonEld commented:
@Patater I removed the extra line and pushed to first commit.
Please review
